presto-main/src/main/java/com/facebook/presto/execution/PrepareTask.java

```diff
@@ -53,7 +58,12 @@ public String explain(Prepare statement)
     @Override
     public CompletableFuture<?> execute(Prepare prepare, TransactionManager transactionManager, Metadata metadata, AccessControl accessControl, QueryStateMachine stateMachine)
     {
-        String sql = getFormattedSql(prepare.getStatement(), sqlParser);
+        Statement statement = prepare.getStatement();
+        if (statement instanceof Prepare || statement instanceof Execute || statement instanceof Deallocate) {
+            throw new PrestoException(INVALID_PREPARED_STATEMENT, "Invalid statement type for prepared statement: " + statement);
```

Comment:
The default `toString()` is not appropriate (or needed) for an end user error message. It might be best to do `statement.getClass().getSimpleName().toUpperCase(ENGLISH)`